src/Remote/Import.php

```diff
@@ -0,0 +1,124 @@
+<?php
+
+namespace Castor\Remote;
+
+use Castor\GlobalHelper;
+use Castor\Remote\Exception\InvalidImportUrl;
+use Castor\Remote\Exception\NotTrusted;
+
+use function Castor\fs;
+use function Castor\get_cache;
+use function Castor\get_input;
+use function Castor\io;
+use function Castor\log;
+use function Castor\run;
+
+/** @internal */
+class Import
+{
+    public static function importFunctions(string $scheme, string $url, bool $dryRun = false): string
+    {
+        if ('github' === $scheme) {
+            if (!preg_match('#^(?<organization>[^/]+)/(?<repository>[^/]+)(?<function_path>[^@]*)@(?<version>.+)$#', $url, $matches)) {
+                throw new InvalidImportUrl('The import path from GitHub repository must be formatted like this: "github://<organization>/<repository>/<function_path>@<version>".');
+            }
+
+            $path = self::importFunctionsFromGitRepository(
+                'github.com',
+                sprintf('%s/%s', $matches['organization'], $matches['repository']),
+                $matches['version'],
+                $matches['function_path'] ?? '/castor.php',
+                $dryRun,
+            );
+
+            log('Using functions from remote resource.', 'info', [
+                'url' => $path,
+            ]);
+
+            return $path;
+        }
+
+        throw new InvalidImportUrl(sprintf('The import scheme "%s" is not supported.', $scheme));
+    }
+
+    private static function importFunctionsFromGitRepository(string $domain, string $repository, string $version, string $functionPath, bool $dryRun): string
+    {
+        self::ensureTrustedResource($domain . '/' . $repository);
+
+        $dir = GlobalHelper::getHomeDirectory() . '/remote/' . $domain . '/' . $repository . '/' . $version;
+
+        if (!is_dir($dir) && !$dryRun) {
+            log("Importing functions in path {$functionPath} from {$domain}/{$repository} (version {$version})");
+
+            fs()->mkdir($dir);
+
+            run(['git', 'clone', "git@{$domain}:{$repository}.git", '--branch', $version, '--depth', '1', '--filter', 'blob:none', '.'], path: $dir, quiet: true);
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❔ question:** ‏Is it possible to add a nice error message for server than don't have "git" by default? (most unix distributions).

Reply:
This is ran on your host, so I guess most developers have git installed :upside_down_face:. Anyway, I guess we can add a check to error gracefully :+1:

Reply:
Done